File: BJ/개인문제풀이/PGM/해결/귤고르기.java
package BJ.개인문제풀이.PGM.해결;

import java.util.*;

public class 귤고르기 {

    public int solution(int k, int[] tangerine) {

        int ans = 0;
        Map<Integer, Integer> map = new HashMap<>();

        for(int i=0; i<tangerine.length; i++){
            map.put(tangerine[i], map.getOrDefault(tangerine[i],0)+1);
        }

        List<Integer> valueList = new ArrayList<>(map.values());
        Collections.sort(valueList, Collections.reverseOrder());

        int sum=0;

        for(int v : valueList){
            if(sum+v >= k){
                ans++;
                break;
            } else{
                sum+=v;
                ans++;
            }
        }

        return ans;



    }

}

File: AlgoPython/문제/귤고르기.py
def solution(k, tangerine):

    ans = 0
    a={}

    for i in tangerine:
        if i in a:
            a[i] +=1
        else:
            a[i] = 1

    a = dict(sorted(a.items(), key=lambda x : x[1], reverse=True))

    for i in a:
        if k<=0:
            return ans
        k-=a[i]
        ans+=1

    return ans
